pkg/apis/cluster/types.go

```diff
@@ -6,26 +6,6 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// ResourceName is the name identifying various resources in a ResourceList.
```

Comment:
I remember why we didn't directly reference `corev1.ResourceName` at first, it's because we considered that there might be some extensions here, different from the `corev1`.

Reply:
Hi @chaunceyjiang, `corev1.ResourceName` is just a `typedef`, not a `enum`, extensions can also assign to `corev1.ResourceName`

There is a demo: https://h3ld32xlpo.feishu.cn/wiki/Rj5mwS3TCigHY4kMKzhcqTggnlg

Reply:
@chaunceyjiang Can you remind me what kind of extensions are here? I totally forgot it.

cc @Poor12

Reply:
https://github.com/karmada-io/karmada/pull/2367#discussion_r946298844